python/dgl/graphbolt/item_sampler.py

```diff
@@ -756,6 +582,18 @@ def __init__(
             )
         self._world_size = dist.get_world_size()
         self._rank = dist.get_rank()
+        device = (
+            torch.cuda.current_device()
+            if torch.cuda.is_available() and dist.get_backend() == "nccl"
+            else "cpu"
+        )
+        if self._rank == 0:
+            seed = np.random.randint(0, np.iinfo(np.int32).max)
+            seed_tensor = torch.tensor(seed, dtype=torch.int32, device=device)
+        else:
+            seed_tensor = torch.empty([], dtype=torch.int32, device=device)
+        dist.broadcast(seed_tensor, src=0)
+        self._seed = seed_tensor.item()
```

Comment:
is it possible to add a dedicated testcase to make sure the collated minibatches are shuffled globally?

Reply:
I'm concerned that since the shuffling are completely random, it is possible that the result will turn out to look like a local shuffling (like how we did before). It is possible to fail such a test, although the probability is very low.

Reply:
then please check if the model accuracy go up or down.

Reply:
Benchmark result updated in the Description. The performance of most examples remain the same.